zrml/simple-disputes/src/lib.rs

```diff
@@ -91,20 +166,92 @@ mod pallet {
         InvalidMarketStatus,
         /// On dispute or resolution, someone tried to pass a non-simple-disputes market type
         MarketDoesNotHaveSimpleDisputesMechanism,
+        StorageOverflow,
+        OutcomeMismatch,
+        CannotDisputeSameOutcome,
+        MarketIsNotReported,
+        MaxDisputesReached,
     }
 
-    #[pallet::event]
-    pub enum Event<T>
-    where
-        T: Config, {}
-
     #[pallet::hooks]
     impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
 
-    impl<T> Pallet<T>
-    where
-        T: Config,
-    {
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight(5000)]
+        #[transactional]
+        pub fn reserve_outcome(
+            origin: OriginFor<T>,
+            #[pallet::compact] market_id: MarketIdOf<T>,
+            outcome: OutcomeReport,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+            let market = T::MarketCommons::market(&market_id)?;
+            ensure!(
+                market.dispute_mechanism == MarketDisputeMechanism::SimpleDisputes,
+                Error::<T>::MarketDoesNotHaveSimpleDisputesMechanism
+            );
+            ensure!(market.status == MarketStatus::Disputed, Error::<T>::InvalidMarketStatus);
+            ensure!(market.matches_outcome_report(&outcome), Error::<T>::OutcomeMismatch);
+            let report = market.report.as_ref().ok_or(Error::<T>::MarketIsNotReported)?;
+
+            let now = <frame_system::Pallet<T>>::block_number();
+            let disputes = Disputes::<T>::get(&market_id);
+            let num_disputes: u32 = disputes.len().saturated_into();
+
+            Self::ensure_can_not_dispute_the_same_outcome(&disputes, report, &outcome)?;
+            Self::ensure_disputes_does_not_exceed_max_disputes(num_disputes)?;
+
+            T::AssetManager::reserve_named(
+                &Self::reserve_id(),
+                Asset::Ztg,
+                &who,
+                default_dispute_bond::<T>(disputes.len()),
+            )?;
+
+            let market_dispute = MarketDispute { at: now, by: who, outcome };
+            <Disputes<T>>::try_mutate(market_id, |disputes| {
+                disputes.try_push(market_dispute.clone()).map_err(|_| <Error<T>>::StorageOverflow)
+            })?;
+
+            // each dispute resets dispute_duration
+            Self::remove_auto_resolve(disputes.as_slice(), &market_id, &market);
+            let dispute_duration_ends_at_block =
+                now.saturating_add(market.deadlines.dispute_duration);
+            T::DisputeResolution::add_auto_resolve(&market_id, dispute_duration_ends_at_block)?;
+
+            Self::deposit_event(Event::OutcomeReserved { market_id, dispute: market_dispute });
+
+            Ok((Some(5000)).into())
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        #[inline]
+        pub fn reserve_id() -> [u8; 8] {
+            T::PredictionMarketsPalletId::get().0
```

Comment:
You could also use migration of the Disputes storage element to use a new reserve identifier just for the dispute bonds derived from the simple disputes pallet id.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/938/commits/6f10d96e2aecce83998dfb7cac26705ad2dc122a